cmd/genautocomplete/genautocomplete_test.go

```diff
@@ -21,6 +21,23 @@ func TestCompletionBash(t *testing.T) {
 	assert.NotEmpty(t, string(bs))
 }
 
+func TestCompletionBashStdout(t *testing.T) {
+	originalStdout := os.Stdout
+	readOut, writeOut, errorPipe := os.Pipe()
```

Comment:
The test failure is caused by the Pipe blocking.

It only has a limited buffer (maybe none) so unless you read what is coming out, you can't add more.

If you run the `bashCommandDefinition.Run(bashCommandDefinition, []string{"-"})` in a go routine (with some synchronisation) then the read and write will proceed in parallel and the pipe will be OK.

Reply:
Looking at https://golang.org/pkg/os/#Pipe, there's no buffer. `os.Pipe()` returns connected `Files`, which I imagine are in-memory file references.

So, after writing to the "in-memory" files, the write file is closed and and read file is read here:
`output, err := ioutil.ReadAll(readOut)`.

I can't imagine why `os.Pipe()` would be blocking just on Windows. Can you?

Reply:
os.Pipe calls syscall.Pipe2 so is using an OS specific pipe mechanism. According the the man page

>  Data  written  to  the write end of the pipe is buffered by the
       kernel until it is read from the read end of the pipe.  For further de‐
       tails, see pipe(7).

So there is some kind of buffer.

> I can't imagine why os.Pipe() would be blocking just on Windows. Can you?

I imagine the buffer is smaller or something like that.

You could redirect stdout to a file instead of using a pipe?